pkg/netceptor/packetconn_test.go

```diff
@@ -0,0 +1,298 @@
+				mockNetceptorForPacketConn.EXPECT().GetListenerRegistry().Return(map[string]*netceptor.PacketConn{})
+			},
+			func(pc netceptor.PacketConner) interface{} {
+				return pc.Close()
+			},
+			nil,
+			"Expected no error, but got %v",
+		},
+		{
+			"Close Error",
+			"test",
+			func(ctx context.Context) {
+				mockNetceptorForPacketConn.EXPECT().Context().Return(context.Background())
+				mockNetceptorForPacketConn.EXPECT().RemoveLocalServiceAdvertisement("test").Return(errors.New("Close Error"))
+				mockNetceptorForPacketConn.EXPECT().GetUnreachableBroker().Return(utils.NewBroker(ctx, reflect.TypeOf(netceptor.UnreachableNotification{})))
+				mockNetceptorForPacketConn.EXPECT().GetListenerLock().Return(&sync.RWMutex{}).Times(2)
+				mockNetceptorForPacketConn.EXPECT().GetListenerRegistry().Return(map[string]*netceptor.PacketConn{})
+			},
+			func(pc netceptor.PacketConner) interface{} {
+				return pc.Close().Error()
+			},
+			"Close Error",
+			"Expected error to be Close Error, but got %v",
+		},
+		{
+			"SetDeadline Success",
+			"test",
+			func(ctx context.Context) {
+				mockNetceptorForPacketConn.EXPECT().Context().Return(context.Background())
+				mockNetceptorForPacketConn.EXPECT().GetUnreachableBroker().Return(utils.NewBroker(ctx, reflect.TypeOf(netceptor.UnreachableNotification{})))
+			},
+			func(pc netceptor.PacketConner) interface{} {
+				return pc.SetDeadline(time.Now().Add(time.Millisecond * 100))
+			},
+			nil,
+			"Expected no error, but got %v",
+		},
+		{
+			"SetReadDeadline Success",
+			"test",
+			func(ctx context.Context) {
+				mockNetceptorForPacketConn.EXPECT().Context().Return(context.Background())
+				mockNetceptorForPacketConn.EXPECT().GetUnreachableBroker().Return(utils.NewBroker(ctx, reflect.TypeOf(netceptor.UnreachableNotification{})))
+			},
+			func(pc netceptor.PacketConner) interface{} {
+				return pc.SetReadDeadline(time.Now().Add(time.Millisecond * 100))
+			},
+			nil,
+			"Expected no error, but got %v",
+		},
+		{
+			"SetWriteDeadline Success",
+			"test",
+			func(ctx context.Context) {
+				mockNetceptorForPacketConn.EXPECT().Context().Return(context.Background())
+				mockNetceptorForPacketConn.EXPECT().GetUnreachableBroker().Return(utils.NewBroker(ctx, reflect.TypeOf(netceptor.UnreachableNotification{})))
+			},
+			func(pc netceptor.PacketConner) interface{} {
+				return pc.SetWriteDeadline(time.Now().Add(time.Millisecond * 100))
+			},
+			nil,
+			"Expected no error, but got %v",
+		},
+	}
+
+	for _, testCase := range packetConnTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			ctx := context.Background()
+			mockNetceptorForPacketConn.EXPECT().MaxForwardingHops().Return(byte(1))
+			mockNetceptorForPacketConn.EXPECT().GetListenerRegistry().Return(map[string]*netceptor.PacketConn{})
+
+			testCase.expectedCall(ctx)
+
+			var returnVal interface{}
+
+			if testCase.name == "Close Error" {
```

Comment:
This `if else` block can be refactored to be more efficient or easier to read.

First off, only 1 test case seems to have `"Close Error"`, and yet its the first check on each loop, meaning that the else is hit for all bar 1 loops.

This means if we reverse the conditions we will same time as the else will only be hit once  `if testCase.name != "Close Error"` 

Taking it a step further, their is only 1 check really, so it would be even less instructions if we just assign `pc` and `returnval` outside of the `if else` block and then on the condition of `"Close Error"` we override the two vars. This would be easier to read and even less instructions.

Reply:
```
pc := netceptor.NewPacketConn(mockNetceptorForPacketConn, testCase.service, 0)
returnVal = testCase.funcCall(pc)
if testCase.name == "Close Error" {
    pc := netceptor.NewPacketConnWithConst(mockNetceptorForPacketConn, testCase.service, true, map[string]string{}, byte(0))
    returnVal = testCase.funcCall(pc)
}
```